### PR TITLE
Client port

### PR DESCRIPTION
**Pull request by lukehinds**
Signed-off-by: Luke Hinds <lhinds@redhat.com>

### CODE REVIEW

File: pkg/oauthflow/pkce.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	}
+
+	var chosenMethod string
+	for _, method := range providerClaims.CodeChallengeMethodsSupported {
+		// per RFC7636, any server that supports PKCE must support S256
+		if method == PKCES256 && chosenMethod != PKCES256 {
+			chosenMethod = PKCES256
+			break
+		} else if method != "plain" {
+			log.Logger.Infof("Unsupported code challenge method in list: '%v'", method)
```
Comment (dlorenc) [inline pkg/oauthflow/pkce.go:56]: I know you just copied this over, but we might want to use the cli.Logger here rather than the production one.

**Review by dlorenc — APPROVED** [review]
Approved, just a few nits that would be fine to handle later with TODOs.

File: pkg/oauthflow/pkce.go (hunk 1 above)
Comment (lukehinds) [reply pkg/oauthflow/pkce.go:56]: yup, totally. makes sense

**Comment by bobcallaway** [comment]
Is there a `Makefile` that can be reused to generate the client stubs?

**Comment by lukehinds** [comment]
> Is there a `Makefile` that can be reused to generate the client stubs?

Yes, I did plan to do that. I was going to talk with you both about how we share the openapi.yml from fulcio into here.

**Comment by lukehinds** [comment]
> Is there a `Makefile` that can be reused to generate the client stubs?

I wanted to chat to you about this, as in how we handle the openapi.yml in this repo. I thought we could either:

Could we do this as a follow up patch? Makes sense to do this as the same time as docker compose files.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,32 @@
+.PHONY: all test clean lint gosec
+
+all: client
```
Comment (lukehinds) [inline Makefile:3]: I guess we don't need this, as there is no 'server' for constituent an 'all'

**Pull request merged**